src/test/kotlin/uk/gov/justice/digital/hmpps/sentenceplan/services/GoalServiceTest.kt

```diff
@@ -250,4 +255,100 @@ class GoalServiceTest {
       assertThat(exception.message).startsWith("One or more of the Related Areas of Need was not found")
     }
   }
+
+  @Nested
+  @DisplayName("UpdateSteps")
+  inner class UpdateSteps {
+    @Test
+    fun `update steps for goal that does not exist should throw an exception`() {
+      every { goalRepository.findByUuid(any()) } returns null
+
+      val exception = assertThrows<Exception> {
```

Comment:
`requires` throws an `IllegalArgumentException`. Can we assert that it throws this instead?

Reply:
We can, but this test won't make it that far. I'll do that below.